src/pystack/_pystack/mem.h

```diff
@@ -109,6 +110,28 @@ class MemoryMapInformation
     std::optional<VirtualMap> d_heap;
 };
 
+class LRUCache
+{
+  private:
+    using list_node_t = std::pair<uintptr_t, size_t>;
```

Comment:
Just a small nit: in general I find expressions like `it->second.second` a bit challenging to check for correctness and it always forces me to go to the definitions. Given that we would not be sacrificing performance or anything I recommend here to use a names struct for `list_node_t` so we can use the names directly instead of the position on the pair.

The same thing applies to `cache_value_t`